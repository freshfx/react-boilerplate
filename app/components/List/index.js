import React, {useMemo} from 'react'
import PropTypes from 'prop-types'

import Ul from './Ul'
import Wrapper from './Wrapper'

const List = ({component: ComponentToRender, items}) => {
  const renderItem = item => (
    <ComponentToRender key={`item-${item}`} id={item} />
  )

  const content = useMemo(() => {
    if (items.length) {
      return items.map(renderItem)
    }
    return <ComponentToRender />
  }, [items])

  return (
    <Wrapper>
      <Ul>{content}</Ul>
    </Wrapper>
  )
}

List.propTypes = {
  component: PropTypes.elementType.isRequired,
  items: PropTypes.array
}

List.defaultProps = {
  items: []
}

export default List
