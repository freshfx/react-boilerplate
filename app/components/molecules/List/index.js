import React from 'react'
import PropTypes from 'prop-types'

import Ul from './Ul'
import Wrapper from './Wrapper'

const List = ({component: ComponentToRender, items}) => {
  const renderItem = item => (
    <ComponentToRender key={`item-${item}`} id={item} />
  )

  return (
    <Wrapper>
      <Ul>
        {(items.length && items.map(renderItem)) || <ComponentToRender />}
      </Ul>
    </Wrapper>
  )
}

List.propTypes = {
  component: PropTypes.elementType.isRequired,
  items: PropTypes.array
}

List.defaultProps = {
  items: []
}

export default List
