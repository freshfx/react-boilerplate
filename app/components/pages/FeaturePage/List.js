import styled from '@emotion/styled'

const List = styled.ul`
  font-family: Georgia, Times, 'Times New Roman', serif;
  padding-left: 1.75em;
`

export default List
