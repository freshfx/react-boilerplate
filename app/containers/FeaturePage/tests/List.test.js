import 'jest-styled-components'
import React from 'react'
import {mount} from 'enzyme'

import List from '../List'

const renderComponent = (props = {}) => mount(<List {...props} />)

describe('<List />', () => {
  it('should match snapshot', () => {
    const renderedComponent = renderComponent()
    expect(renderedComponent).toMatchSnapshot()
  })
})

