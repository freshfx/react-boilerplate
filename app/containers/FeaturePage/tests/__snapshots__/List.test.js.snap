// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match snapshot 1`] = `
.c0 {
  font-family: Georgia,Times,'Times New Roman',serif;
  padding-left: 1.75em;
}

<List>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "List-sc-1aai6b-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "font-family:Georgia,Times,'Times New Roman',serif;padding-left:1.75em;",
          ],
        },
        "displayName": "List",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "List-sc-1aai6b-0",
        "target": "ul",
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <ul
      className="c0"
    />
  </StyledComponent>
</List>
`;
